src/main/target/WARPF7/target.h

```diff
@@ -0,0 +1,141 @@
+
+#define TARGET_BOARD_IDENTIFIER     "WARP"
+#define USBD_PRODUCT_STRING         "WARPF7"
+
+// Beeper
+#define USE_BEEPER
+#define BEEPER                      PA15
+#define BEEPER_INVERTED
+
+// Leds
+#define USE_LED_STRIP
+#define WS2811_PIN                  PA8
+#define LED0                        PC3
+#define LED1                        PB2
+
+// UARTs
+#define USB_IO
+#define USE_VCP
+#define USE_UART1
+#define UART1_RX_PIN                PA10
+#define UART1_TX_PIN                PA9
+#define USE_UART2
+#define UART2_RX_PIN                PA3
+#define UART2_TX_PIN                PA2
+#define USE_UART3
+#define UART3_RX_PIN                PC11
+#define UART3_TX_PIN                PC10
+#define USE_UART4
+#define UART4_RX_PIN                PA1
+#define UART4_TX_PIN                PA0
+#define USE_UART5
+#define UART5_RX_PIN                PD2
+#define UART5_TX_PIN                PC12
+#define SERIAL_PORT_COUNT           6
+#define DEFAULT_RX_TYPE             RX_TYPE_SERIAL
+#define SERIALRX_PROVIDER           SERIALRX_CRSF
+
+// SPI
+#define USE_SPI
+#define USE_SPI_DEVICE_1
+#define SPI1_SCK_PIN                PA5
+#define SPI1_MISO_PIN               PA7
+#define SPI1_MOSI_PIN               PA6
+#define USE_SPI_DEVICE_2
+#define SPI2_SCK_PIN                PB13
+#define SPI2_MISO_PIN               PB15
+#define SPI2_MOSI_PIN               PB14
+#define USE_SPI_DEVICE_3
+#define SPI3_SCK_PIN                PB3
+#define SPI3_MISO_PIN               PB5
+#define SPI3_MOSI_PIN               PB4
+
+// I2C
+#define USE_I2C
+#define USE_I2C_DEVICE_1
+#define I2C1_SCL                    PB8
+#define I2C1_SDA                    PB9
+
+// MAG
+#define USE_MAG
+#define USE_MAG_ALL
+#define MAG_I2C_BUS                 BUS_I2C1
+
+// ADC
+#define ADC_CHANNEL_1_PIN           PC2
+#define VBAT_ADC_CHANNEL            ADC_CHN_1
+#define ADC_CHANNEL_2_PIN           PC1
+#define CURRENT_METER_ADC_CHANNEL   ADC_CHN_2
+#define ADC_CHANNEL_3_PIN           PC0
+#define RSSI_ADC_CHANNEL            ADC_CHN_3
+#define USE_ADC
+#define ADC_INSTANCE ADC1
+
+
+// Gyro & ACC
+#define USE_IMU_ICM42605
+#define IMU_ICM42605_ALIGN          CW270_DEG
+#define ICM42605_CS_PIN             PA4
+#define ICM42605_SPI_BUS            BUS_SPI1
+
+// BARO
+#define USE_BARO
+#define USE_BARO_BMP280
+#define USE_BARO_SPI_BMP280
+#define BMP280_SPI_BUS              BUS_SPI3
+#define BMP280_CS_PIN               PC13
+
+// OSD
+#define USE_MAX7456
+#define MAX7456_CS_PIN              PB12
+#define MAX7456_SPI_BUS             BUS_SPI2
+
+// Blackbox
+#define USE_FLASHFS
+#define ENABLE_BLACKBOX_LOGGING_ON_SPIFLASH_BY_DEFAULT
+#define USE_FLASH_M25P16
+#define M25P16_SPI_BUS              BUS_SPI3
+#define M25P16_CS_PIN               PB6
+
+// PINIO
+#define USE_PINIO
+#define USE_PINIOBOX
+#define PINIO1_PIN                  PC14
+#define PINIO2_PIN                  PC15
+
+// Others
+#define MAX_PWM_OUTPUT_PORTS        8
+#define USE_SERIAL_4WAY_BLHELI_INTERFACE
+#define USE_DSHOT
```

Comment:
Dshot is not working, you also need #define USE_DSHOT_DMAR.

With USE_DSHOT_DMAR motor 2 is still not working. Probably need to adjust options in target.c